kola/tests/podman/podman.go

```diff
@@ -0,0 +1,352 @@
+	Image  string `json:"image"`
+	Status string `json:"status"`
+}
+
+// simplifiedPsInfo represents the results of podman ps -a
+type simplifiedPsInfo struct {
+	containers []simplifiedContainerPsInfo
+}
+
+// simplifiedPodmanInfo represents the results of podman info
+type simplifiedPodmanInfo struct {
+	Store struct {
+		GraphDriverName string `json:"GraphDriverName"`
+		GraphRoot       string `json:"GraphRoot"`
+	}
+}
+
+func getSimplifiedPsInfo(c cluster.TestCluster, m platform.Machine) (simplifiedPsInfo, error) {
+	target := simplifiedPsInfo{}
+	psJson, err := c.SSH(m, `sudo podman ps -a --format json`)
+
+	if err != nil {
+		return target, fmt.Errorf("could not get info: %v", err)
+	}
+
+	err = json.Unmarshal(psJson, &target.containers)
+
+	if err != nil {
+		return target, fmt.Errorf("could not unmarshal info %q into known json: %v", string(psJson), err)
+	}
+	return target, nil
+}
+
+// Returns the result of podman info as a simplifiedPodmanInfo
+func getPodmanInfo(c cluster.TestCluster, m platform.Machine) (simplifiedPodmanInfo, error) {
+	target := simplifiedPodmanInfo{}
+
+	pInfoJson, err := c.SSH(m, `sudo podman info --format json`)
+	if err != nil {
+		return target, fmt.Errorf("Could not get info: %v", err)
+	}
+
+	err = json.Unmarshal(pInfoJson, &target)
+	if err != nil {
+		return target, fmt.Errorf("Could not unmarshal info %q into known JSON: %v", string(pInfoJson), err)
+	}
+	return target, nil
+}
+
+// make a podman container out of binaries on the host
+func genPodmanContainer(c cluster.TestCluster, m platform.Machine, name string, binnames []string) {
+	cmd := `tmpdir=$(mktemp -d); cd $tmpdir; echo -e "FROM scratch\nCOPY . /" > Dockerfile;
+	        b=$(which %s); libs=$(sudo ldd $b | grep -o /lib'[^ ]*' | sort -u);
+	        sudo rsync -av --relative --copy-links $b $libs ./;
+	        sudo podman build -t %s .`
+
+	c.MustSSH(m, fmt.Sprintf(cmd, strings.Join(binnames, " "), name))
+}
+
+func podmanBaseTest(c cluster.TestCluster) {
+	c.Run("info", podmanInfo)
+	c.Run("resources", podmanResources)
+	c.Run("network", podmanNetworksReliably)
+}
+
+// Test: Run basic podman commands
+func podmanWorkflow(c cluster.TestCluster) {
+	m := c.Machines()[0]
+
+	// Test: Verify container can run with volumen mount and port forwarding
+	image := "docker.io/library/nginx"
+	wwwRoot := "/usr/share/nginx/html"
+
+	dir := c.MustSSH(m, `mktemp -d`)
+
+	cmd := fmt.Sprintf("echo TEST PAGE > %s/index.html", string(dir))
+	c.MustSSH(m, cmd)
+
+	cmd = fmt.Sprintf("sudo podman run -d -p 80:80 -v %s/index.html:%s/index.html:z %s", string(dir), wwwRoot, image)
+	out := c.MustSSH(m, cmd)
+	id := string(out)
+
+	podIsRunning := func() error {
+		b, err := c.SSH(m, `curl -f http://localhost 2>/dev/null`)
+		if err != nil {
+			return err
+		}
+		if !bytes.Contains(b, []byte("TEST PAGE")) {
+			return fmt.Errorf("nginx pod is not running %s", b)
+		}
+		return nil
+	}
+
+	if err := util.Retry(10, 10*time.Second, podIsRunning); err != nil {
```

Comment:
Does this really need 100+ seconds? Can we bump this down to 30 seconds with something like `util.Retry(6, 5*time.Second, podIsRunning)`?

Reply:
Probably not.  I will bump down to 30 seconds.